server/server.go

```diff
@@ -381,14 +382,14 @@ func (s *Server) Run(host, token, nodeName string, insecure bool) error {
 	if s.MetricsPort == s.AppPort {
 		r.Handle("/metrics", metrics.GetHandler())
 	} else {
-		metrics.StartMetricsServer(s.MetricsPort)
+		metrics.StartMetricsServer(s.BindIP, s.MetricsPort)
```

Comment:
What do you think of using a dedicated listener ? Because if the agent isn't hostNetworked we can't fetch its metrics.

Reply:
We could do that, yes.
Let's test it and add it right after